README.md

```diff
@@ -47,7 +47,7 @@ CREATE SEQUENCE user_auth.auth_id_seq
 CREATE TABLE user_auth.auth (
   id integer DEFAULT nextval('user_auth.auth_id_seq'::regclass) NOT NULL,
   salt text NOT NULL,
-  token text NOT NULL,
+  token text,
```

Comment:
I'm guessing this `token` is the session identifier. Not a fan that it lives inside what it should be the `users` table.
You should consider creating a `sessions` table with a Foreign Key to this table.

sessions
id|token|user_id|created_at

As a general rule of thumb, think of responsibilities and ownership. 
Is the _User_ responsibility to set their own password? Yes, they're the ones that have to remember it.
Is the _User_ responsibility to manage their own session? No, it's ours and it should be transparent to them. This means it's likely that there's other behavior attached to this that we're not seeing yet (as a hint, I'll ask you to expire the session at some point, but not yet).

Reply:
great insight thanks, I added the `sessions` table